src/Engine/EmitMSIL/CodeGenIL.cs

```diff
@@ -285,8 +308,18 @@ private Type EmitIListCoercion<T>()
         }
 
         // Coerce int/long/double arrays to IEnumerable<T> or IList<T>
-        private Type EmitArrayCoercion<Source, Target>(Type ienumerableParamType)
+        private Type EmitArrayCoercion<Source, Target>(AssociativeNode arg,Type ienumerableParamType)
         {
+            if (compilePass == CompilePass.GatherTypeInfo)
+            {
+                var returnType = typeof(Target[]);
+                if (typeof(IList<Target>).IsAssignableFrom(ienumerableParamType))
```

Comment:
I think this should be changed from `IList` to `List` (I made the same mistake earlier) as `IList<T>` is assignable from `T[]`.